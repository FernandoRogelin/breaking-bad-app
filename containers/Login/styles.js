import styled, { css } from 'styled-components';

import ContentComponent from '#/components/Content';

export const Section = styled.section`
  width: auto;
  display: flex;
  justify-content: center;
  padding: 36px 14px 15px;
`;

export const Content = styled(ContentComponent)`
  width: 835px;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Form = styled.form`
  width: auto;
  margin: 0 auto;
  max-width: 585px;
  padding-bottom: 30px;
`;

export const Title = styled.h1`
  margin: 20px 0 25px;
`;

export const Input = styled.input`
  width: 97%;
  outline: none;
  padding: 4px;
  font-size: 14px;
  max-width: 280px;
  line-height: 20px;
  border-radius: 3px;
  margin-bottom: 15px;
  background-color: #fff;
  border: 1px solid #e6ecf0;

  ${({ error }) =>
    error &&
    css`
      border: 1px solid red;
    `}

  &::placeholder {
    color: #8899a6;
  }
`;
