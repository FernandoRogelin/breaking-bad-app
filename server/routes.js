const routes = require('next-routes');

module.exports = routes()
  .add('login')
  .add('episodes')
  .add('characters')
  .add('quotes');
