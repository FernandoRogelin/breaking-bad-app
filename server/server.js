const Koa = require('koa');
const Router = require('koa-router');
const helmet = require('koa-helmet');
const logger = require('koa-logger');
const bodyParcer = require('koa-bodyparser');
const nextjs = require('next');

const nextRoutes = require('./routes');
const port = process.env.PORT || 8080;
const app = nextjs({ dev: process.env.NODE_ENV !== 'production' });
const nextHandler = nextRoutes.getRequestHandler(app);

app
  .prepare()
  .then(() => {
    const server = new Koa();
    const router = new Router();

    server
      .use(logger())
      .use(bodyParcer())
      .use(helmet());

    router.get('/', ctx => {
      ctx.redirect('/login');
    });

    // Handle NextJS pages
    router.get('*', async ctx => {
      await nextHandler(ctx.req, ctx.res);

      ctx.status = 200;
      ctx.respond = false;
    });

    server.use(router.routes(), router.allowedMethods());

    server.listen(port, err => {
      if (err) console.log(err);
      console.log(`Server running in port http://localhost:${port}`);
    });
  })
  .catch(ex => {
    console.log(ex.stack);
    process.exit(1);
  });
