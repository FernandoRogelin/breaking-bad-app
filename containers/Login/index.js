import React from 'react';
import { Router } from '#/server/routes';

import * as Yup from 'yup';
import { Formik } from 'formik';
import Conta from '#/conta';
import Button from '#/components/Button';

import { Section, Wrapper, Title, Input, Form, Content } from './styles';

export default function Login() {
  const schema = Yup.object().shape({
    email: Yup.string()
      .email()
      .required('Campo obrigatório'),
    password: Yup.string().required('Campo obrigatório'),
  });

  function handleSubmit(values) {
    if (Conta.login === values.email && Conta.senha === values.password) {
      Router.pushRoute('timeline');
    }
  }

  return (
    <Section>
      <Content>
        <Formik
          initialValues={{
            email: '',
            password: '',
          }}
          onSubmit={values => handleSubmit(values)}
          validationSchema={schema}
        >
          {({
            dirty,
            values,
            errors,
            touched,
            isValid,
            handleBlur,
            handleChange,
            handleSubmit,
          }) => (
            <Form onSubmit={handleSubmit}>
              <Title>Entrar no site</Title>
              <Wrapper>
                <Input
                  id="email"
                  name="email"
                  onBlur={handleBlur}
                  placeholder="Login"
                  value={values.email}
                  onChange={handleChange}
                  data-testid="input-email"
                  error={errors.email && touched.email}
                />
                <Input
                  id="password"
                  name="password"
                  type="password"
                  onBlur={handleBlur}
                  placeholder="Password"
                  value={values.password}
                  onChange={handleChange}
                  data-testid="input-password"
                  error={errors.password && touched.password}
                />
              </Wrapper>
              <Button
                data-testid="button-submit"
                disabled={!isValid || !dirty}
                type="submit"
              >
                Enviar
              </Button>
            </Form>
          )}
        </Formik>
      </Content>
    </Section>
  );
}
