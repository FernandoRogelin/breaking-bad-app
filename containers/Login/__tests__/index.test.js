import React from 'react';
import { render, fireEvent, wait } from 'react-testing-library';
import { Router } from '#/server/routes';

import Login from '../';

const handleSubmit = jest.fn();
Router.pushRoute = jest.fn();

describe('Login container', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Snapshot test', () => {
    const { container } = render(<Login />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('Inicial flow', async () => {
    const { getByTestId } = render(<Login />);

    expect(getByTestId('button-submit')).toBeDisabled();

    await wait(() => {
      fireEvent.change(getByTestId('input-email'), {
        target: { value: 'teste@hotmail.com' },
      });
    });

    await wait(() => {
      fireEvent.change(getByTestId('input-password'), {
        target: { value: '123456' },
      });
    });

    await wait(() => fireEvent.click(getByTestId('button-submit')));

    await wait(() => expect(handleSubmit).toHaveBeenCalledTimes(0));
  });
});
