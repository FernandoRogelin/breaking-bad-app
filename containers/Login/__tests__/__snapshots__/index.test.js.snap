// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container Snapshot test 1`] = `
<section
  class="styles__Section-vyh5mw-0 cpPGWb"
>
  <div
    class="Content-vb879o-0 styles__Content-vyh5mw-1 tpVMy"
  >
    <form
      class="styles__Form-vyh5mw-3 jIFoNM"
    >
      <h1
        class="styles__Title-vyh5mw-4 bCEKEo"
      >
        Entrar no site
      </h1>
      <div
        class="styles__Wrapper-vyh5mw-2 kxYQQc"
      >
        <input
          class="styles__Input-vyh5mw-5 bUdEyG"
          data-testid="input-email"
          id="email"
          name="email"
          placeholder="Login"
          value=""
        />
        <input
          class="styles__Input-vyh5mw-5 bUdEyG"
          data-testid="input-password"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        class="Button-n3cyj4-0 gTwdqt"
        data-testid="button-submit"
        disabled=""
        type="submit"
      >
        Enviar
      </button>
    </form>
  </div>
</section>
`;
